src/foundry/client/data/abstract/client-document.d.mts

```diff
@@ -329,11 +329,10 @@ declare class ClientDocument<
    * Gets the default new name for a Document
    * @param context - The context for which to create the Document name.
    */
-  static defaultName<T extends DocumentConstructor>(
+  static defaultName(
     context?: InexactPartial<{
-      this: T;
```

Comment:
So I wasn't actually asking for `this` parameter to be removed entirely. Did you decide it wasn't useful?

Reply:
It's not doing anything until someone adds in the `type` handling, so I figured I'd remove it for now then when the TODO is addressed it can be added back in.